crypto/sm2/sm2_pmeth.c

```diff
@@ -32,12 +32,24 @@ typedef struct {
 static int pkey_sm2_init(EVP_PKEY_CTX *ctx)
 {
     SM2_PKEY_CTX *smctx;
+    uint8_t *id = SM2_DEFAULT_USERID;
+    size_t id_len = strlen(id);
+    uint8_t *tmp_id;
 
     if ((smctx = OPENSSL_zalloc(sizeof(*smctx))) == NULL) {
         SM2err(SM2_F_PKEY_SM2_INIT, ERR_R_MALLOC_FAILURE);
         return 0;
     }
 
+    tmp_id = OPENSSL_malloc(id_len);
```

Comment:
replace lines 44-49 with OPENSSL_strdup ?

Reply:
OPENSSL_strdup need more than once strlen invoketion

Reply:
Optimizing a strlen compared with code clarity is the wrong thing to do.

Reply:
got it!